### ISSUE TITLE
🎉 Release for Sprint 228 (v3.4.0) 🎉 

### ISSUE THREAD

**Issue by rm3l** [redhat-developer/odo#6262]
Issue to track work for publishing a new release of `odo` at the end of the Sprint.

- Target release process start date: **Dec 12, 2022**
- Errata date (when we expect the binaries to be published): **Dec 15, 2022**
- Version: **v3.4.0**.
  - Since this is not a pre-release, make sure to update the [`build/VERSION`](https://github.com/redhat-developer/odo/blob/main/build/VERSION) file  (only after binaries are available on the Content Gateway)
- Once release is done, make sure link to download `latest` points to v3.4.0: https://developers.redhat.com/content-gateway/rest/mirror/pub/openshift-v4/clients/odo/latest/

Notes:
- Write an announcement blog post once the release is done
- Document the following changes in the release notes / blog post:
  - `odo list services -o json`: `group` field removed and superseded by a new `apiVersion` field (see https://github.com/redhat-developer/odo/pull/6358)

/kind task
/area release-eng
